test/test_distributions.py

```diff
@@ -748,6 +748,22 @@ def test_distribution_constraints(jax_dist, sp_dist, params, prepend_shape):
         d.log_prob(oob_samples)
 
 
+def test_omnistaging_invalid_param():
```

Comment:
We have tests that check whether arg constraints are satisfied, but I suppose we do not have tests that check validation for out of support values in distributions? If so, it will be good to add tests for all relevant distributions and do a patch release. WDYT?

Reply:
This test is fine, but my concern is if we have missed something like this for other distributions.

Reply:
Yeah, I think we already have some tests that check for both constraints and support. Let me also add `jit` check for them.

Reply:
Actually, you raised a very good point. It is just lucky that my test passed for `positive` constraint. For many other constraints, no warning is raised! :( The issue is if the constraint uses `jax.numpy` to check, the output will be a traced array (under jit), regardless the input is a DeviceArray or not.